### PR TITLE
20 backsetup apollo

### PR DESCRIPTION
**Pull request by Cavonstavant**
This PR add basic setup of apollo in coordination of a mongodb database.
It also setup a first User type and Query in Apollo

- db: Add readme
- db: Add package.json and yarn.lock
- db: Add basic prisma configuration along with graphql and apollo setup
- db: Add bson to needed packages
- db: Add yarn.lock
- orm: Add schema generation command in REAME
- orm: Add `@default(auto())` to `User.id` field

### CODE REVIEW

File: database/prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    provider = "prisma-client-js"
+}
+
+datasource db {
+    provider = "mongodb"
+    url      = env("MONGODB_URI")
+}
+
+model User {
+    id        String   @default(auto()) @id @db.ObjectId @map("_id")
```
Comment (aurelenc) [inline database/prisma/schema.prisma:11]: ?

Comment (aurelenc) [reply database/prisma/schema.prisma:11]: @id @default(uuid())

File: database/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+![](https://badgen.net/badge/icon/graphql?icon=graphql&label) Typescript project with GraphQL and Apollo
+
+## Prisma
+
+### Generate prisma schema in prisma client (only for seeding the db)
+```bash
+npx prisma generate --schema=./prisma/schema.prisma
+```
+
+> Require a running mongodb instance running on localhost
```
Comment (EmileEttel) [inline database/README.md:12]:
```suggestion
> Requires a mongodb instance to be running on localhost
```

**Review by EmileEttel — COMMENTED** [review]
LGTM (except the readme)

File: database/src/app.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { ApolloServer } from 'apollo-server';
+import { resolvers, typeDefs } from './schemas/index';
+import { context } from './context';
+import { env } from './config/env';
+
+new ApolloServer({ resolvers, typeDefs, context }).listen({ port: env.APOLLO_PORT }, () => {
+    console.log('🚀 [' + env.ENV_NAME + '] Server is running at ' + env.APOLLO_HOST + ':' + env.APOLLO_PORT);
```
Comment (aurelenc) [inline database/src/app.ts:7]:
```suggestion
    console.log('[' + env.ENV_NAME + '] Server is running at ' + env.APOLLO_HOST + ':' + env.APOLLO_PORT);
```

**Review by EmileEttel — APPROVED** [review]

**Review by aurelenc — APPROVED** [review]

**Review by Bootoyka — COMMENTED** [review]
wp ntrfm

**Pull request merged**